tests/test_lookahead_iterator.py

```diff
@@ -17,27 +18,32 @@ def test_lookahead_iterator(self):
         self.assertEqual(cursor.goto_first_child(), True)  # struct
         self.assertEqual(cursor.goto_first_child(), True)  # struct keyword
 
-        next_state = cursor.node.next_parse_state
+        node = cast(Node, cursor.node)
+        next_state = node.next_parse_state
 
         self.assertNotEqual(next_state, 0)
         self.assertEqual(
-            next_state, self.rust.next_state(cursor.node.parse_state, cursor.node.grammar_id)
+            next_state, self.rust.next_state(node.parse_state, node.grammar_id)
         )
         self.assertLess(next_state, self.rust.parse_state_count)
         self.assertEqual(cursor.goto_next_sibling(), True)  # type_identifier
-        self.assertEqual(next_state, cursor.node.parse_state)
-        self.assertEqual(cursor.node.grammar_name, "identifier")
-        self.assertNotEqual(cursor.node.grammar_id, cursor.node.kind_id)
+        node = cast(Node, cursor.node)
+        self.assertEqual(next_state, node.parse_state)
+        self.assertEqual(node.grammar_name, "identifier")
+        self.assertNotEqual(node.grammar_id, node.kind_id)
 
         expected_symbols = ["//", "/*", "identifier", "line_comment", "block_comment"]
-        lookahead = self.rust.lookahead_iterator(next_state)
+        lookahead = cast(LookaheadIterator, self.rust.lookahead_iterator(next_state))
         self.assertEqual(lookahead.language, self.rust)
-        self.assertListEqual(list(lookahead.iter_names()), expected_symbols)
+        self.assertListEqual(lookahead.names(), expected_symbols)
```

Comment:
>_possible bug_

**✨ AI Suggestion:** The `cast` here seems unnecessary as `self.rust.lookahead_iterator(next_state)` already returns a `LookaheadIterator` object. Consider removing the cast for better readability.
```suggestion
        lookahead = self.rust.lookahead_iterator(next_state)
        self.assertEqual(lookahead.language, self.rust)
        self.assertListEqual(lookahead.names(), expected_symbols)
```